sdk/search/search/src/searchIndexClient.ts

```diff
@@ -447,15 +448,14 @@ export class SearchIndexClient<T> {
     documents: T[],
     options: UploadDocumentsOptions = {}
   ): Promise<IndexDocumentsResult> {
-    const actionType = options.mergeIfExists ? "mergeOrUpload" : "upload";
     const { span, updatedOptions } = createSpan("SearchIndexClient-uploadDocuments", options);
 
-    const batch = documents.map<IndexAction<T>>((doc) => {
-      return {
-        ...doc,
-        actionType
-      };
-    });
+    const batch = new IndexDocumentsBatch<T>();
+    if (options.mergeIfExists) {
```

Comment:
With the batching operations now in place, should there really be options like this and `uplaodIfNotExists` below?

Reply:
I'm fine ditching these and simply adding `mergeOrUploadDocuments` on the convenience layer to replace them.